Lesson_40/project/src/App.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useEffect, useState } from 'react';
+import Header from './Header/Header';
+import { Routes, Route } from "react-router-dom";
+import CreateProduct from './CreateProduct/CreateProduct';
+import Products from './Products/Products';
+import './App.css';
+
+export const NotificationContext = createContext();
+
+const App = () => {
+  const [products, setProducts] = useState([]);
+  const [productsInCart, setProductsInCart] = useState([]);
+
+  const addNewProduct = (product) => {
+    setProducts([...products, product]);
+    console.log(products);
+  }
+
+  const addProductToCart = (product) => {
+    setProductsInCart([...productsInCart, product]);
+    console.log(productsInCart); 
+  }
+
+  const apiGet = () => {
+    fetch('https://dummyjson.com/products')
+    .then((res) => res.json())
+    .then((data) => setProducts(data.products))
+  };
+
+  useEffect(() => {
+    apiGet();
+  }, []);
+
+  return (
+    <>
+    <NotificationContext.Provider value={[products, productsInCart, addProductToCart]}>
```

Comment:
better to pass it by object then
what if user wants a third `addProductToCart` but isn't interested of the other
he will forced to import everything